manual/src/cmds/lexyacc.etex

```diff
@@ -724,4 +724,96 @@ positions inside the scanned lexer buffer. The size of this table is
 limited to at most 255 cells. This error should not show up in normal
 situations.
 
+\item[ocamlyacc: concurrency safety]
+
+Parsers generated by ocamlyacc are not generally concurrency safe.
+Those parsers rely on an internal work state which is specific to each
+domain but is shared between various threads.
+
+It is safe to spawn multiple parsers in separate domains and let them run to completion.
+For instance, one may use the small "calc" parser defined in the example section, in two
+domains:
+\begin{verbatim}
+let parse s () =
+  let lexbuf = Lexing.from_string s in
+  let r = Parser.main Lexer.token lexbuf in
+  Format.printf "%d@." r
+
+let () =
+  let dl = List.map Domain.spawn [parse "1+6\n"; parse "7+8\n"; parse "9+17\n"] in
+  List.iter Domain.join dl
+\end{verbatim}
+
+
+It is also safe to suspend and resume a parser on the same domain if no other parsers run in the interim on this domain.
+
+\begin{verbatim}
+let parse s =
+  let lexbuf = Lexing.from_string s in
+  let r = Parser.main Lexer.token lexbuf in
+  Format.printf "%d@." r
+
+let rec print s n =
+  if n = 0 then ()
+  else begin
+    Format.printf "print(%s) %d@." s n;
+    Thread.yield ();
+    print s (n-1)
+  end
+
+let on_domain s () =
+  let t1 = Thread.create (print s) 10 in
+  let t2 = Thread.create parse (s^"\n") in
+  Thread.join t1; Thread.join t2
+
+let () =
+  let dl =
+    List.map (fun s ->Domain.spawn (on_domain s))
+      ["1+6"; "7+8"; "9+17"]
+  in
+  List.iter Domain.join dl
+\end{verbatim}
+
+
+It is not safe to migrate a parser execution to a different domain or to interleave the execution of two or more parsers on the same domain.
```

Comment:
I'm not sure all our users understand that "interleave the execution" includes in particular "using the Threads library".